File: java/json-transform/src/main/java/co/nlighten/jsontransform/functions/TransformerFunctionYaml.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.functions.common.FunctionContext;
import co.nlighten.jsontransform.functions.common.TransformerFunction;
import co.nlighten.jsontransform.formats.yaml.YamlFormat;

public class TransformerFunctionYaml extends TransformerFunction {

    public TransformerFunctionYaml() {
        super();
    }
    @Override
    public Object apply(FunctionContext context) {
        return new YamlFormat(context.getAdapter()).serialize(context.getUnwrapped(null, true));
    }
}

File: javascript/json-transform/src/functions/TransformerFunctionYaml.ts
import TransformerFunction from "./common/TransformerFunction";
import FunctionContext from "./common/FunctionContext";
import YamlFormat from "../formats/yaml/YamlFormat";

class TransformerFunctionYaml extends TransformerFunction {
  constructor() {
    super({});
  }

  override async apply(context: FunctionContext): Promise<any> {
    return YamlFormat.INSTANCE.serialize(await context.getUnwrapped(null, true));
  }
}

export default TransformerFunctionYaml;
